knownvalue: Add `TupleExact`, `TuplePartial` and `TupleSizeExact` checks for dynamic value testing

austinvalle opened the pull request:
Closes #312

Exact copies of the `List*` equivalents.

### Notes

I explicitly didn't add documentation, since these are only really useful for dynamic value testing, but I can add them if we think it's necessary 👍🏻

bflad reviewed (approved):
Looks good to me 🚀 

I think it might be worth the additional tuple documentation page, even if it is copy-paste-modify of the list page for now (maybe with a small callout that tuples are relatively uncommon except for testing dynamic types).

austinvalle commented:
> I think it might be worth the additional tuple documentation page, even if it is copy-paste-modify of the list page for now (maybe with a small callout that tuples are relatively uncommon except for testing dynamic types).

Added in 29d51b5 ⚡ - I adjusted the wording as well as added a little blurb mentioning they are only encountered when testing dynamic data 🚀
